core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -1096,33 +1100,41 @@ private[spark] class Executor(
     )
 
     if (useStub) {
-      createClassLoaderWithStub(urls, conf.get(CONNECT_SCALA_UDF_STUB_PREFIXES))
+      createClassLoaderWithStub(urls, conf.get(CONNECT_SCALA_UDF_STUB_PREFIXES), isDefaultSession)
     } else {
-      createClassLoader(urls)
+      createClassLoader(urls, isDefaultSession)
     }
   }
 
-  private def createClassLoader(urls: Array[URL]): MutableURLClassLoader = {
+  private def createClassLoader(urls: Array[URL],
+                                isDefaultSession: Boolean): MutableURLClassLoader = {
+    // SPARK-51537, The isolation session must "inherit" the classloader from default session which
+    // has already added the global jars specified by --jars
```

Comment:
For sake of completeness/tracking, please also add the short blurb about the plugin issue here and the other comment below

Reply:
Done. Thx